### PR TITLE
Added an 'output' options group

### PR DESCRIPTION
**Pull request by jlstevens**
This PR builds on #3492 in order to tackle the suggestion in #3463: tracking backend in options should be done consistently with everything else by introducing an 'output' options group.

### CODE REVIEW

**Comment by jlstevens** [comment]
Note that since commit 226d65005, you can do this:

image

Deciding whether we want to use this flexibility to control renderers is a point for discussion. I just wanted to point out that this is now possible.

**Comment by jlstevens** [comment]
I think this PR is now ready to review/merge.

File: holoviews/tests/core/testdimensioned.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ def register_custom(cls, objtype, backend, custom_plot=[], custom_style=[]):
         style_opts = Keywords(['style_opt1', 'style_opt2']+custom_style, name)
         plot_opts = Keywords(['plot_opt1', 'plot_opt2']+custom_plot, name)
         opt_groups = {'plot': Options(allowed_keywords=plot_opts),
-                      'style': Options(allowed_keywords=style_opts)}
+                      'style': Options(allowed_keywords=style_opts),
+                      'output':Options(allowed_keywords=['backend'])}
```
Comment (philippjfr) [inline holoviews/tests/core/testdimensioned.py]:
```suggestion
                      'output': Options(allowed_keywords=['backend'])}
```

File: holoviews/core/util.py [hunk 2]
```diff
@@ -242,7 +242,7 @@ def deprecated_opts_signature(args, kwargs):
     """
     from .options import Options
     groups = set(Options._option_groups)
-    opts = {kw for kw in kwargs if kw not in ('backend', 'clone')}
+    opts = {kw for kw in kwargs if kw !='clone'}
```
Comment (philippjfr) [inline holoviews/core/util.py]:
```suggestion
    opts = {kw for kw in kwargs if kw != 'clone'}
```

File: holoviews/core/dimension.py [hunk 3]
```diff
@@ -1388,7 +1388,7 @@ def options(self, *args, **kwargs):
         Returns:
             Returns the cloned object with the options applied
         """
-        backend = kwargs.pop('backend', None)
+        backend = kwargs.get('backend', None)
```
Comment (philippjfr) [inline holoviews/core/dimension.py]: Now that the backend is still in kwargs won't the ``elif args and kwargs:`` branch execute and error?

File: holoviews/core/dimension.py [hunk 4]
```diff
@@ -1406,18 +1406,25 @@ def options(self, *args, **kwargs):
                              "or explicitly define the type, e.g. "
                              "obj.options({'Image': {'cmap': 'viridis'}}). "
                              "Supplying both formats is not supported.")
+        elif args and isinstance(args[0], dict):
+            options = [Options(spec, **kws) for spec,kws in args[0].items()]
```
Comment (philippjfr) [inline holoviews/core/dimension.py]: If there is more than one arg this should either warn or process all the args, silently dropping other args doesn't seem great.

File: holoviews/core/options.py [hunk 5]
```diff
@@ -158,6 +158,18 @@ def __call__(self, *args, **kwargs):
         Returns:
             Returns the object or a clone with the options applied
         """
+        if self._mode is None:
+            apply_groups, _, _ = deprecated_opts_signature(args, kwargs)
+            if apply_groups and config.future_deprecations:
+                msg = ("Calling the .opts method with options broken down by options "
```
Comment (philippjfr) [inline holoviews/core/options.py]: Confused this is in ``__call__`` but is warning about ``.opts``?

File: holoviews/core/dimension.py (hunk 3 above)
Comment (jlstevens) [reply holoviews/core/dimension.py]: I don't think there is any issue here - the point of the PR is that `backend` is now like any other kwarg. It is just that the value is needed here (if available) for `obj._dispatch_opts`.

File: holoviews/core/options.py [hunk 6]
```diff
@@ -224,13 +236,6 @@ def _base_opts(self, *args, **kwargs):
 
         # By default do not clone in .opts method
         clone = kwargs.get('clone', None)
-
-        if apply_groups and config.future_deprecations:
-            msg = ("Calling the .opts method with options broken down by options "
-                   "group (i.e. separate plot, style and norm groups) is deprecated. "
-                   "Use the .options method converting to the simplified format "
-                   "instead or use hv.opts.apply_groups for backward compatibility.")
-            param.main.warning(msg)
```
Comment (philippjfr) [inline holoviews/core/options.py]: Is there now any future warning for the deprecated formats? I know there was the issue with the recursive calls which meant it was always triggered when enabled but I had hoped there was a way around this.

File: holoviews/core/dimension.py (hunk 3 above)
Comment (philippjfr) [reply holoviews/core/dimension.py]: It will now error when you supply args and the backend as a kwarg as far as I can tell.

Comment (philippjfr) [reply holoviews/core/dimension.py]: e.g. as far as I can tell ``.options(opts.Image(cmap='Reds'), backend='matplotlib')`` will get caught up in the ``elif args and kwargs`` branch and raise a ValueError. Maybe I'm missing something?

File: holoviews/core/dimension.py (hunk 4 above)
Comment (jlstevens) [reply holoviews/core/dimension.py]: Agreed though the use of `args[0]` wasn't introduced in this PR. I'll add the warnings as suggested.

File: holoviews/core/options.py (hunk 5 above)
Comment (philippjfr) [reply holoviews/core/options.py]: Nevermind, I realize this is the .opts accessor object.

File: holoviews/core/options.py (hunk 6 above)
Comment (philippjfr) [reply holoviews/core/options.py]: Nevermind, same confusion as the previous comment about ``__call__``. I can't visualize who calls what anymore, I'd really need a diagram to see how the different formats get converted and then pencil in the warnings at the appropriate stages.

File: holoviews/core/dimension.py (hunk 4 above)
Comment (jlstevens) [reply holoviews/core/dimension.py]: Unfortunately, it isn't simple to fix due to the way `opts` and `options` invoke each other. Valid signatures would issue such a warning.

File: holoviews/core/dimension.py (hunk 3 above)
Comment (jlstevens) [reply holoviews/core/dimension.py]: I don't think `.options` is supposed to accept `Option` objects afaik. Only `.opts` should.

File: holoviews/core/dimension.py (hunk 4 above)
Comment (philippjfr) [reply holoviews/core/dimension.py]: Not sure I follow, when would passing in multiple dictionaries as kwargs ever occur?

File: holoviews/core/dimension.py (hunk 3 above)
Comment (philippjfr) [reply holoviews/core/dimension.py]: I didn't think that was true, I thought they were basically equivalent except for clone and that ``.options`` doesn't allow the really old nested format. In any case though the point I'm making, the same problem applies to ``.opts({'Image': dict(cmap='Reds')}, backend='matplotlib')``

Comment (jlstevens) [reply holoviews/core/dimension.py]: Addressed in f112875

**Comment by philippjfr** [comment]
Looks good will merge once passing.

File: holoviews/core/dimension.py (hunk 4 above)
Comment (jlstevens) [reply holoviews/core/dimension.py]: I can probably warn for this one condition when dicts are involved but not for other cases where `args[0]` is used and potentially dropping other positional arguments.

Comment (jlstevens) [reply holoviews/core/dimension.py]: Addressed in 9b9240bc1

**Pull request merged**